javalib/src/main/scala/java/util/IdentityHashMap.scala

```diff
@@ -0,0 +1,297 @@
+              iter.remove()
+              removeAll(iter, true)
+            } else
+              removeAll(iter, modified)
+          } else
+            modified
+        }
+        removeAll(this.iterator, false)
+      }
+
+    override def retainAll(c: Collection[_]): Boolean = {
+      val iter = iterator()
+      var changed = false
+      while (iter.hasNext) {
+        val elem = iter.next()
+        if (!findSame(elem, c)) {
+          iter.remove()
+          changed = true
+        }
+      }
+      changed
+    }
+
+    override def clear(): Unit =
+      self.clear()
+  }
+
+  private final class EntrySet extends AbstractSet[Map.Entry[K, V]] {
+    def iterator(): Iterator[Map.Entry[K, V]] = {
+      new ju.Iterator[Map.Entry[K, V]] {
+        private val iter = inner.entrySetIterator()
+
+        def hasNext(): Boolean =
+          iter.hasNext()
+
+        def next(): Map.Entry[K, V] =
+          new MapEntry(iter.next())
+
+        def remove(): Unit =
+          iter.remove()
+      }
+    }
+
+    def size(): Int =
+      inner.size()
+
+    override def contains(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          self.containsKey(thatKey) &&
+            same(self.get(thatKey), value.getValue())
+        case _ =>
+          false
+      }
+
+    override def remove(value: Any): Boolean =
+      value match {
+        case value: Map.Entry[_, _] =>
+          val thatKey = value.getKey()
+          val thatValue = value.getValue()
+          if (self.containsKey(thatKey) && same(self.get(thatKey), thatValue))
+              same(self.remove(thatKey), thatValue)
+          else
+            false
+        case _ =>
+          false
+      }
+
+    override def clear(): Unit =
+      inner.clear()
+  }
+}
+
+object IdentityHashMap {
+  private final case class IdentityBox[+K](inner: K) {
+    override def equals(o: Any): Boolean = {
+      o match {
+        case o: IdentityBox[_] =>
+          // inner.asInstanceOf[AnyRef] eq o.inner.asInstanceOf[AnyRef]
+          // Use the following until 1.0.0
+          // See https://github.com/scala-js/scala-js/pull/3747
+          same(inner, o.inner)
+        case _ =>
+          false
+      }
+    }
+
+    override def hashCode(): Int =
+      System.identityHashCode(inner)
+  }
+
+  @inline private def same(v1: Any, v2: Any): Boolean = {
+    val v1Ref = v1.asInstanceOf[AnyRef]
+    val v2Ref = v2.asInstanceOf[AnyRef]
+    if (v1Ref eq v2Ref)
+      (v1Ref ne java.lang.Double.valueOf(0.0)) ||
+        (1.0 / v1Ref.asInstanceOf[Double] == 1.0 / v2Ref.asInstanceOf[Double]) // +0.0 v -0.0
+    else
+      (v1Ref ne v1Ref) && (v2Ref ne v2Ref) // NaN
+  }
+
+  private def findSame[K](elem: K, c: Collection[_]): Boolean = {
+    val iter = c.iterator()
+    var found = false
+    while (iter.hasNext && !found) {
+      if (same(elem, iter.next()))
+        found = true
```

Comment:
You can revert this to using `return true` with the appropriate Scalastyle commands.